src/java/org/apache/cassandra/locator/TokenMetadata.java

```diff
@@ -1039,6 +1041,13 @@ public Set<InetAddress> getAllEndpoints()
         }
     }
 
+    public Set<InetAddress> getAllMembers()
```

Comment:
I think since tMD is the source of truth for the gossiper, we should do something similar to isMember (which is how the gossiper determines membership) and consult the tokenToEndpointMap.